Issue starting the App serices

I followed all the instructions. All my steps complete successfully. I get all deployment but when I click on App Service URL, getting following error 
#######################################
:( Application Error
If you are the application administrator, you can access the [diagnostic resources]

######################################



<!--
IF SUFFICIENT INFORMATION IS NOT PROVIDED VIA THE FOLLOWING TEMPLATE THE ISSUE MIGHT BE CLOSED WITHOUT FURTHER CONSIDERATION OR INVESTIGATION
-->
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>

### Any log messages given by the failure
>

### Expected/desired behavior
>

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)

### azd version?
> run `azd version` and copy paste here.

### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

I have same problem when opening App Service URL. Sometimes I get "504.0 GatewayTimeout", but mostly it's 

*:( Application Error
If you are the application administrator, you can access the [diagnostic resources]*

I followed tips for debug. There are errors In default_docker.log and it never gets to the point `[INFO] Application startup complete`

2024-04-11T06:31:29.7372552Z [2024-04-11 06:31:29 +0000] [74] [INFO] Starting gunicorn 21.2.0
2024-04-11T06:31:29.7410245Z [2024-04-11 06:31:29 +0000] [74] [INFO] Listening at: http://0.0.0.0:8000 (74)
2024-04-11T06:31:29.7999465Z [2024-04-11 06:31:29 +0000] [74] [INFO] Using worker: uvicorn.workers.UvicornWorker
2024-04-11T06:31:29.8015967Z [2024-04-11 06:31:29 +0000] [75] [INFO] Booting worker with pid: 75
2024-04-11T06:31:29.8378628Z [2024-04-11 06:31:29 +0000] [76] [INFO] Booting worker with pid: 76
2024-04-11T06:31:37.7735807Z [2024-04-11 06:31:29 +0000] [77] [INFO] Booting worker with pid: 77
2024-04-11T06:31:38.2815899Z [2024-04-11 06:31:30 +0000] [78] [INFO] Booting worker with pid: 78
2024-04-11T06:31:38.2838019Z [2024-04-11 06:31:30 +0000] [79] [INFO] Booting worker with pid: 79
2024-04-11T06:35:48.1419482Z [2024-04-11 06:35:22 +0000] [74] [CRITICAL] WORKER TIMEOUT (pid:75)
2024-04-11T06:35:49.1810074Z [2024-04-11 06:35:47 +0000] [74] [CRITICAL] WORKER TIMEOUT (pid:76)
2024-04-11T06:35:49.3895327Z [2024-04-11 06:35:47 +0000] [74] [CRITICAL] WORKER TIMEOUT (pid:77)
2024-04-11T06:35:49.3901006Z [2024-04-11 06:35:47 +0000] [74] [CRITICAL] WORKER TIMEOUT (pid:78)
2024-04-11T06:35:49.3906297Z [2024-04-11 06:35:47 +0000] [74] [CRITICAL] WORKER TIMEOUT (pid:79)
2024-04-11T06:35:49.7671480Z [2024-04-11 06:35:49 +0000] [74] [ERROR] Worker (pid:76) was sent SIGKILL! Perhaps out of memory?
2024-04-11T06:35:49.7854543Z [2024-04-11 06:35:49 +0000] [74] [ERROR] Worker (pid:77) was sent SIGKILL! Perhaps out of memory?
2024-04-11T06:35:49.9497755Z [2024-04-11 06:35:49 +0000] [74] [ERROR] Worker (pid:78) was sent SIGKILL! Perhaps out of memory?
2024-04-11T06:35:49.9912356Z [2024-04-11 06:35:49 +0000] [74] [ERROR] Worker (pid:75) was sent SIGKILL! Perhaps out of memory?
2024-04-11T06:35:50.0784409Z [2024-04-11 06:35:50 +0000] [74] [ERROR] Worker (pid:79) was sent SIGKILL! Perhaps out of memory?

### Other details:
- I'm mostly using free SKU tiers for the resources
- Running app locally works just fine

--- a/app/backend/gunicorn.conf.py
+++ b/app/backend/gunicorn.conf.py
@@ -1,4 +1,5 @@
 import multiprocessing
+import os
 
 max_requests = 1000
 max_requests_jitter = 50
@@ -9,5 +10,9 @@ timeout = 230
 # https://learn.microsoft.com/en-us/troubleshoot/azure/app-service/web-apps-performance-faqs#why-does-my-request-time-out-after-230-seconds
 
 num_cpus = multiprocessing.cpu_count()
-workers = (num_cpus * 2) + 1
+if os.getenv("WEBSITE_SKU") == "LinuxFree":
+    # Free tier reports 2 CPUs but can't handle multiple workers
+    workers = 1
+else:
+    workers = (num_cpus * 2) + 1
 worker_class = "uvicorn.workers.UvicornWorker"

